### DISCUSSION TITLE
ImageSharp V3 should target .NET 6 Only

### DISCUSSION THREAD

**Discussion by JimBobSquarePants** [SixLabors/ImageSharp#2072] [Ideas]
I've been putting a lot of thought to it and in order to avoid the increasing developmental distance between .NET Standard and .NET our V3 release should target .NET 6 will which be the current LTS release.

Using a .NET Standard target would mean that they cannot use any of the [Runtime Intrinsics APIs](https://apisof.net/catalog/4dce742d-4168-7767-02ef-f4fb6734b652). Adding an additional target to the old LTS .NET Core 3.1 would mean adding a target that reaches EOL in [December 2022](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core). I can't see us shipping V3 until after then.

Users currently targeting older frameworks will always have v2.1.0 to fall back on. 

@SixLabors/core if any of you have any objection to this please let me know.

**Comment by dlemstra** [comment]
I do agree that this would make the code easier to maintain. There are a lot of `#if` checks in the code. And I also think that this project highly benefits from using the Runtime Intrinsic API. I am missing .NET Framework in your message but looking at the title I suspect that we only want to use .NET 6 (or other LTS release) going forward. I have no objections against this. Maybe we should do this for all our repositories?

**Reply by JimBobSquarePants** [reply]
Yeah, definitely no .NET Framework. LTS and future majors would also always target the latest LTS. All others would eventually follow suite once we get stable releases out.

**Comment by antonfirsov** [comment]
No objections as long as we setup a proper patching process for ImageSharp `2.1.*`. By that I mean committing to deliver/backport security fixes and critical bugfixes. No features or minor bugfixes unless a commercial customer is directly funding them.

**Reply by JimBobSquarePants** [reply]
Yep agreed. There’s already release branches to branch from to create bug fix releases.
